### ISSUE TITLE
[!] Your current version ($(MARKETING_VERSION)) does not respect the format A or A.B or A.B.C

### ISSUE THREAD

**Issue by SaundersB** [fastlane/fastlane#15676]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues) <-- It was closed automatically.

### Issue Description
Attempting to update XCode project version. I.E. A minor bump version fail such as 1.44 to 1.45.
I am able to successfully update the build number like build 345 to 346 but I cannot update the version number. There was a related issue that was closed due to inactivity to this problem.

##### Complete output when running fastlane, including the stack trace and command used
<details>
  <pre>$ fastlane bump
[✔] 🚀 
+----------------------------+---------+--------------------------------------------------------------------+
|                                               Used plugins                                                |
+----------------------------+---------+--------------------------------------------------------------------+
| Plugin                     | Version | Action                                                             |
+----------------------------+---------+--------------------------------------------------------------------+
| fastlane-plugin-versioning | 0.4.0   | get_version_number_from_git_branch                                 |
|                            |         | increment_build_number_in_xcodeproj get_version_number_from_plist  |
|                            |         | get_build_number_from_plist ci_build_number                        |
|                            |         | get_version_number_from_xcodeproj                                  |
|                            |         | increment_version_number_in_xcodeproj                              |
|                            |         | increment_version_number_in_plist get_app_store_version_number     |
|                            |         | get_info_plist_path get_build_number_from_xcodeproj                |
|                            |         | increment_build_number_in_plist                                    |
+----------------------------+---------+--------------------------------------------------------------------+

[13:51:15]: ------------------------------
[13:51:15]: --- Step: default_platform ---
[13:51:15]: ------------------------------
[13:51:15]: Driving the lane 'ios bump' 🚀
[13:51:15]: --------------------------------------
[13:51:15]: --- Step: increment_version_number ---
[13:51:15]: --------------------------------------
[13:51:15]: Before being able to increment and read the version number from your Xcode project, you first need to setup your project properly. Please follow the guide at https://developer.apple.com/library/content/qa/qa1827/_index.html
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios bump |
+------------------+----------+
[13:51:15]: Your current version ($(MARKETING_VERSION)) does not respect the format A or A.B or A.B.C

+------+--------------------------+-------------+
|               fastlane summary                |
+------+--------------------------+-------------+
| Step | Action                   | Time (in s) |
+------+--------------------------+-------------+
| 1    | default_platform         | 0           |
| 💥   | increment_version_number | 0           |
+------+--------------------------+-------------+

[13:51:15]: fastlane finished with errors

[!] Your current version ($(MARKETING_VERSION)) does not respect the format A or A.B or A.B.C
</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.1                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.21.0 (Apple Git-122.2)                           |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.1 (19B88)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.2.1                                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane


default_platform(:ios)

platform :ios do
    
 
  lane :bump do
    increment_version_number(xcodeproj: "<project>.xcodeproj", bump_type: "minor")
  end

  lane :increment do
      increment_build_number(xcodeproj: "<project>.xcodeproj")
  end

  lane :release do
    desc "Push a new release build to the iOS App Store"
    ENV['ENVFILE'] = '.env.production'
    increment
    pod
    build
    upload_to_app_store()
    commit
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("org.<company>.<project>") # The bundle identifier of your app
apple_id("bsaunders@<company>.org") # Your Apple email address

itc_team_id("<numbers>") # App Store Connect Team ID
team_id("<dev_stuff>") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.136.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                     | Version | Update-Status |
| -------------------------- | ------- | ------------- |
| fastlane-plugin-versioning | 0.4.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                        | Version      |
| -------------------------- | ------------ |
| did_you_mean               | 1.3.0        |
| slack-notifier             | 2.3.2        |
| atomos                     | 0.1.3        |
| claide                     | 1.0.3        |
| colored2                   | 3.1.2        |
| nanaimo                    | 0.2.6        |
| xcodeproj                  | 1.13.0       |
| rouge                      | 2.0.7        |
| xcpretty                   | 0.3.0        |
| terminal-notifier          | 2.0.0        |
| unicode-display_width      | 1.6.0        |
| terminal-table             | 1.8.0        |
| plist                      | 3.5.0        |
| public_suffix              | 2.0.5        |
| addressable                | 2.7.0        |
| multipart-post             | 2.0.0        |
| word_wrap                  | 1.0.0        |
| tty-screen                 | 0.7.0        |
| tty-cursor                 | 0.7.0        |
| tty-spinner                | 0.9.1        |
| babosa                     | 1.0.3        |
| colored                    | 1.2          |
| highline                   | 1.7.10       |
| commander-fastlane         | 4.4.6        |
| excon                      | 0.68.0       |
| faraday                    | 0.17.0       |
| unf_ext                    | 0.0.7.6      |
| unf                        | 0.1.4        |
| domain_name                | 0.5.20190701 |
| http-cookie                | 1.0.3        |
| faraday-cookie_jar         | 0.0.6        |
| faraday_middleware         | 0.13.1       |
| fastimage                  | 2.1.7        |
| gh_inspector               | 1.1.3        |
| mini_magick                | 4.9.5        |
| multi_xml                  | 0.6.0        |
| rubyzip                    | 1.3.0        |
| security                   | 0.1.3        |
| xcpretty-travis-formatter  | 1.0.0        |
| dotenv                     | 2.7.5        |
| bundler                    | 1.17.2       |
| naturally                  | 2.2.0        |
| simctl                     | 1.6.6        |
| jwt                        | 2.1.0        |
| uber                       | 0.1.0        |
| declarative                | 0.0.10       |
| declarative-option         | 0.1.0        |
| representable              | 3.0.4        |
| retriable                  | 3.1.2        |
| mime-types-data            | 3.2019.1009  |
| mime-types                 | 3.3          |
| multi_json                 | 1.14.1       |
| signet                     | 0.12.0       |
| memoist                    | 0.16.0       |
| os                         | 1.0.1        |
| googleauth                 | 0.6.7        |
| httpclient                 | 2.8.3        |
| google-api-client          | 0.23.9       |
| google-cloud-env           | 1.3.0        |
| google-cloud-core          | 1.4.1        |
| digest-crc                 | 0.4.1        |
| google-cloud-storage       | 1.16.0       |
| emoji_regex                | 1.0.1        |
| CFPropertyList             | 3.0.1        |
| json                       | 2.2.0        |
| forwardable                | 1.2.0        |
| logger                     | 1.3.0        |
| stringio                   | 0.0.2        |
| ipaddr                     | 1.2.2        |
| openssl                    | 2.1.2        |
| ostruct                    | 0.1.0        |
| strscan                    | 1.0.0        |
| date                       | 2.0.0        |
| fileutils                  | 1.1.0        |
| etc                        | 1.0.1        |
| io-console                 | 0.4.7        |
| zlib                       | 1.0.0        |
| libxml-ruby                | 3.1.0        |
| rexml                      | 3.1.9        |
| psych                      | 3.1.0        |
| mutex_m                    | 0.1.0        |
| fastlane-plugin-versioning | 0.4.0        |
</details>

*generated on:* **2019-11-18**
</details></pre>
</details>

**Comment by Incorporalis** [comment]
Hi, @SaundersB `build_settings["MARKETING_VERSION"]` in `build_configurations.first` should show you the version number, so if it's nil, check the old way.

**Comment by SaundersB** [comment]
Where is the build_configurations file?

**Comment by joshdholtz** [comment]
Fixed by #15737 💪 This will be in the next release!
